src/classes/UTIL_Profile.cls

```diff
@@ -0,0 +1,92 @@
+/*
+    Copyright (c) 2017, Salesforce.org
+    All rights reserved.
+    
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+    
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+ 
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2017
+* @group Utilities
+* @description Utility to Profiles in the Org
+*/
+public with sharing class UTIL_Profile {
+	/** @description static instance of the current class (UTIL_Profile). It is used as the instance in a Singleton context*/
+	private static UTIL_Profile utilProfile;
+	/** @description Map of the Profiles grouped by Name*/
+	private Map<String,List<Profile>> profilesByName;
+
+	/*********************************************************************************************************
+    * @description get the instance of the current class
+    * @return UTIL_Profile The instance of the current class.
+    */
+	public static UTIL_Profile getInstance() {
+		if (utilProfile == null) {
+			utilProfile = new UTIL_Profile();
+		}
+		return utilProfile;
+	}
+
+	/*********************************************************************************************************
+    * @description Constructor. The Constructor is private because we are using the Singleton Pattern.
+    * The UTIL_Profile object needs to be constructed only from the static method getInstance
+    */
+	private UTIL_Profile() {
+		List<Profile> profileRecords = [
+			SELECT Name
+			FROM Profile
+		];
+		profilesByName = getProfilesByName(profileRecords);
```

Comment:
No need to separate Profile SOQL from the `getProfilesByName`, ie the query can be moved into the `getProfilesByName`.  The method can be moved just after the constructor so related methods are grouped together in the class (for better readability).
 
Moreover, DB queries or any heavy-duty work is not recommended in constructors; however, since this is a Singleton and the constructor is private, it can stay as it is.